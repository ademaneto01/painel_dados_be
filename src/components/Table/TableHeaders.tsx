interface TableHeadersProps {
  headers: string[];
}

export default function TableHeaders(props: TableHeadersProps) {
  return (
    <thead>
      <tr>
        {props.headers.map((header) => (
          <th key={header}>{header}</th>
        ))}
      </tr>
    </thead>
  );
}
