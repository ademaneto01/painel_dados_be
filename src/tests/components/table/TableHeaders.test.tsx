import React from 'react';
import { render } from '@testing-library/react';
import TableHeaders from '../../../components/Table/TableHeaders';

describe('TableHeaders', () => {
  it('renders headers correctly', () => {
    const headers = ['Nome', 'Idade', 'Cargo'];

    const { getByText } = render(
      <table>
        <TableHeaders headers={headers} />
      </table>,
    );

    headers.forEach((header) => {
      const headerElement = getByText(header);
      expect(headerElement).toBeInTheDocument();
    });
  });

  it('renders the correct number of headers', () => {
    const headers = ['Nome', 'Idade', 'Cargo'];

    const { container } = render(
      <table>
        <TableHeaders headers={headers} />
      </table>,
    );

    const thElements = container.querySelectorAll('th');
    expect(thElements.length).toBe(headers.length);
  });

  it('matches the snapshot', () => {
    const headers = ['Nome', 'Idade', 'Cargo'];

    const { asFragment } = render(
      <table>
        <TableHeaders headers={headers} />
      </table>,
    );

    expect(asFragment()).toMatchSnapshot();
  });
});
