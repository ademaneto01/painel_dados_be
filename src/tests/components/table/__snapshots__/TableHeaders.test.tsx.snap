// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TableHeaders matches the snapshot 1`] = `
<DocumentFragment>
  <table>
    <thead>
      <tr>
        <th>
          Nome
        </th>
        <th>
          Idade
        </th>
        <th>
          Cargo
        </th>
      </tr>
    </thead>
  </table>
</DocumentFragment>
`;
